pynamodb/attributes.py

```diff
@@ -979,6 +979,28 @@ def as_dict(self):
             result[key] = value.as_dict() if isinstance(value, MapAttribute) else value
         return result
 
+    # some of the dict API
+
+    def get(self, item: _KT, default: Any = None) -> Union[_VT, None]:
+        if not self._is_attribute_container():
+            raise AssertionError("MapAttribute must be acting as attribute container to MapAttribute.get")
```

Comment:
Perhaps use TypeError like in similar cases in this class?

Reply:
Will do